learningequality/studio
# Add quick edit modals for Boolean Maps Properties

## Summary
### Description of the change(s) you made

Add quick edit modals for:
* Categories
* Learning Activities
* What you will need (refactor)
* Levels

These modals were very similar so this PR introduces a general [EditBooleanMapModal](https://github.com/learningequality/studio/pull/4386/files#diff-72df5874edae701b693eeea0ad95d3d9fa88b5da72de205a355fc3cd33ac44b1) component to put common functionality in one place and then only call it from other components with different props. With this we can reproduce a heriachical options view as the required for [categories modal](https://github.com/learningequality/studio/issues/3416), dynamically add an autocomplete input and recieves validators if needed.

### Screenshots

https://github.com/learningequality/studio/assets/51239030/31e06ace-1daa-4798-ad9c-9d87e4902f8f

___
## Reviewer guidance
### How can a reviewer test these changes?
1. Go to update a channel content
2. Click on the edit described properties on the content node menu (edit learning activities just appears for resources).
3. Or you can select multiple content nodes and select any of the edit properties on the bulk toolbar (edit activities just appears if all selected nodes are resources).


## References
Closes #3416
Closes #3417
Closes #3418

----

### Contributor's Checklist


PR process:


Studio-specifc:



Testing:

___

### Reviewer's Checklist
#### This section is for reviewers to fill out.
